app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Price/Query/BaseFinalPrice.php

```diff
@@ -222,7 +222,7 @@ public function getQuery(array $dimensions, string $productType, array $entityId
 
         if ($entityIds !== null) {
             if (count($entityIds) > 1) {
-                $select->where(sprintf('e.entity_id BETWEEN %s AND %s', min($entityIds), max($entityIds)));
+                $select->where('e.entity_id IN(?)', $entityIds);
```

Comment:
@kandy @mslabko @duhon do you remember? Something similar was done in other place. Please share your thoughts.

Reply:
Yay, I met it accidentally: https://github.com/magento/magento2/pull/19878#discussion_r243217689